torch/distributed/_composable/fully_shard.py

```diff
@@ -30,7 +30,6 @@
 def fully_shard(
     module: nn.Module,
     process_group: Optional[dist.ProcessGroup] = None,
-    sharding_strategy: Optional[ShardingStrategy] = None,
     mixed_precision: Optional[MixedPrecision] = None,
```

Comment:
shall we be conservative on adding the args below as well? E.g., can/should replicate/fully_shard share the same mixed_precision API/design? Similar for cpu offload. I assume we can also apply cpu offload for replicate? ignored_modules/param_init_fn/sync_module_states might all need some more discussion.

Reply:
Yes, let us discuss the arguments in an upcoming meeting.

I think we need to be careful for `replicate` with CPU offload. I discussed briefly with @rohan-varma, and he mentioned that we want to avoid offloading `world_size`-many copies of the parameters. He had an idea where each rank only offloads `1 / world_size` of the parameters from GPU to some shared CPU memory and each rank reads _all_ parameters from CPU to GPU. I am not sure if that can work in general though.